common/src/main/scala/akka/contrib/persistence/mongodb/MongoReadJournal.scala

```diff
@@ -72,9 +73,12 @@ class ScalaDslMongoReadJournal(impl: MongoPersistenceReadJournallingApi)(implici
 
   def allEvents(offset: Long): Source[EventEnvelope, NotUsed] = {
     val pastSource = impl.currentAllEvents(offset)
-    val realtimeSource = Source.actorRef(100, OverflowStrategy.dropTail).mapMaterializedValue(impl.subscribeJournalEvents)
+    val realtimeSource = Source.actorRef[Event](100, OverflowStrategy.dropTail).mapMaterializedValue(ref => {
+                                                                                  impl.subscribeJournalEvents(ref)
+                                                                                  NotUsed
+                                                                                })
     (pastSource ++ realtimeSource)
-      .filter(_.timestamp >= offset)
+      .filter(_.timestamp > offset)
```

Comment:
I see your point, but according to the spec the offset should be unique, this is why I should be exclusive. 

I agree with you that the mere timestamp is too naive, and your proposal is the way to go.

But as now, using the mere timestamp exclusively is too dangerous we are going to lose messages especially because we don't have the timestamp up to ms for most of the events.